### PR TITLE
Release 1.7.0

### PR DESCRIPTION
**Pull request by gnugat**
* Removed `File` deprecations
* Extended public API to extension interfaces
* Deprecated `LineNumberSearchStrategy` and `PhpSearchStrategy`

Once this PR has been reviewed, I'll release it as being the version 1.7.0, and update our sublime couscous documentation :) .

Big thanks to @WouterJ for the contributions, and to @pyrech for the reviews!

### CODE REVIEW

File: UPGRADE-2.0.md [hunk 1]
```diff
@@ -35,7 +35,11 @@ The following exceptions have been removed:
...
 `Gnugat\Redaktilo\Exception`.
 
-## File removal (1.6)
+## PHP and Number Search Removal (1.7)
 
-* `File` has been removed, use `Text` instead
-* `Editor#save`'s `filename` argument is now mandatory
+The following classes have been removed:
+
+* `LineNumberSearchStrategy`
```
Comment (pyrech) [inline UPGRADE-2.0.md]: Maybe mention how to achieve this, maybe like this you did on the `@deprecated` phpdoc of the class:
`Use Text#setCurrentLineNumber instead`

File: src/Gnugat/Redaktilo/Search/PhpSearchStrategy.php [hunk 2]
```diff
@@ -32,6 +34,7 @@ public function __construct(TextToPhpConverter $converter)
     /** {@inheritdoc} */
     public function findAbove(Text $text, $pattern, $location = null)
     {
+        trigger_error(__CLASS__.' is no longer supported and will be removed in version 2', \E_USER_DEPRECATED);
```
Comment (pyrech) [inline src/Gnugat/Redaktilo/Search/PhpSearchStrategy.php]: Did you forgot to add the trigger in the `findBelow` method?

File: src/Gnugat/Redaktilo/Search/LineNumberSearchStrategy.php [hunk 3]
```diff
@@ -15,12 +15,15 @@
...
  * Moves x lines above or below the current one.
+ *
+ * @deprecated 1.7 Use Text#setCurrentLineNumber instead
  */
 class LineNumberSearchStrategy implements SearchStrategy
 {
     /** {@inheritdoc} */
     public function findAbove(Text $text, $pattern, $location = null)
     {
+        trigger_error(__CLASS__.' has been replaced by Text#setCurrentLineNumber', \E_USER_DEPRECATED);
```
Comment (pyrech) [inline src/Gnugat/Redaktilo/Search/LineNumberSearchStrategy.php]: Same here for the findBelow method

**Comment by gnugat** [comment]
@pyrech thanks for the review, I've taken it into account :)

**Comment by pyrech** [comment]
\o/

**Pull request merged**